src/utils.cpp

```diff
@@ -738,6 +738,45 @@ size_t utils::wcswidth_stfl(const std::wstring& str, size_t size) {
 	return width - reduce_count;
 }
 
+std::string utils::substr_with_width(
+		const std::string& str,
+		const size_t max_width)
+{
+	if (str.empty() || !max_width) {
```

Comment:
I'd rather see `max_width == 0` instead of `!max_width`. It's simply more readable.